src/game/suggest.go

```diff
@@ -43,6 +43,26 @@ func Taxicab(grid Grid, grid2 Grid) int {
 	return sum
 }
 
+func TaxicabWithValues(grid Grid, grid2 Grid) int {
+	sum := 0
+	size := len(grid)
+	y := 0
+
+	for y < size {
+		x := 0
+		for x < size {
+			if grid[y][x] != grid2[y][x] {
+				expectedPos, _ := findTileByValue(grid2, grid[y][x])
+				sum += int(math.Abs(float64(y-int(expectedPos.Y))) + math.Abs(float64(x-int(expectedPos.X))))
+				sum += size*size - int(grid[y][x])
```

Comment:
Why is this line for?

Reply:
It's a way to prioritize tiles with low value.

The float64 conversion is for the Math.Abs method.